cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -103,12 +105,16 @@ func (m *Manager) Deploy(ctx context.Context) (*DeployResult, error) {
 	if !filepath.IsAbs(infraRoot) {
 		infraRoot = filepath.Join(m.projectPath, m.options.Path)
 	}
-	bindMountOperations, err := azdFileShareUploadOperations(infraRoot, *m.env)
+	model, err := azdOperations(infraRoot, *m.env)
+	if err != nil {
+		return nil, err
+	}
+	bindMountOperations, err := azdFileShareUploadOperations(model)
 	azdOperationsEnabled := m.alphaFeatureManager.IsEnabled(AzdOperationsFeatureKey)
 	if !azdOperationsEnabled && len(bindMountOperations) > 0 {
 		m.console.Message(ctx, ErrBindMountOperationDisabled.Error())
 	}
-	if azdOperationsEnabled {
+	if azdOperationsEnabled && len(bindMountOperations) > 0 {
```

Comment:
I'm not sure when this azd operations was introduced but we already have a similar concept and we need to standardize on a single method for invoking these types of things.

For example, both `project.ProjectConfig` and `project.ServiceConfig` support event dispatching and we have a few instances that are already leveraging this feature:

1. AKS service target: Registers `postprovision` and `predeploy` operations to ensure the `kubectl` context is set before invoking any k8s commands.

2. Container App Service target: Register `preprovision` operation to ensure the `EXISTS` variables are set to support proper upsert behavior for container apps.

In my opinion we should leverage this feature since is already exposes all the azd lifecycle hooks for both project and services.  Lets chat.